src/iuse_actor.cpp

```diff
@@ -2565,6 +2602,27 @@ repair_item_actor::attempt_hint repair_item_actor::repair( player &pl, item &too
         roll = NEUTRAL;
     }
 
+    if( !requirements.empty() ) {
+        const inventory &crafting_inv = pl.crafting_inventory();
+        const int req_amount = fix.volume() / 125_ml;
+        for( auto const &requirement : requirements ) {
+            auto req_id = requirement_id( requirement );
```

Comment:
This copy is pointless, `requirement` is already an requirement id. Just use it directly or change the loop variable to `req_id`.